Sources_v3/Workers/MessageUpdater_Tests.swift

```diff
@@ -0,0 +1,559 @@
+        }
+        
+        // Try to edit another user's message
+        var completionCalledError: Error?
+        messageUpdater.editMessage(messageId: anotherUserMessagePayload.id, text: .unique) {
+            completionCalledError = $0
+        }
+        
+        // Assert `MessageCanNotBeUpdatedByCurrentUser` is received
+        AssertAsync.willBeTrue(completionCalledError is ClientError.MessageCanNotBeUpdatedByCurrentUser)
+    }
+
+    func test_editMessage_updatesLocalMessageCorrectly_ifThereIsNoLocalState() throws {
+        let updatedText: String = .unique
+
+        // Create current user and channel in the database
+        let (_, cid) = try createCurrentUserAndChannel()
+        
+        // Create a new message in the database
+        let createdMessageId = try createMessage(in: cid, localState: nil)
+
+        // Edit created message with new text
+        var completionCalled = false
+        messageUpdater.editMessage(messageId: createdMessageId, text: updatedText) {
+            XCTAssertNil($0)
+            completionCalled = true
+        }
+        
+        // Load created message
+        var message: MessageDTO? {
+            database.viewContext.message(id: createdMessageId)
+        }
+        
+        AssertAsync {
+            // Assert completion is called
+            Assert.willBeTrue(completionCalled)
+            // Assert message has `pendingSync` state
+            Assert.willBeEqual(message?.localMessageState, .pendingSync)
+            // Assert message has updated text
+            Assert.willBeEqual(message?.text, updatedText)
+        }
+    }
+    
+    func test_editMessage_updatesLocalMessageCorrectly_ifLocalStateIsPendingSync() throws {
+        let updatedText: String = .unique
+
+        // Create current user and channel in the database
+        let (_, cid) = try createCurrentUserAndChannel()
+        
+        // Create a new message in the database with `pendingSync` local state
+        let createdMessageId = try createMessage(in: cid, localState: .pendingSync)
+
+        // Edit created message with `updatedText`
+        var completionCalled = false
+        messageUpdater.editMessage(messageId: createdMessageId, text: updatedText) {
+            XCTAssertNil($0)
+            completionCalled = true
+        }
+        
+        // Load the message
+        var message: MessageDTO? {
+            database.viewContext.message(id: createdMessageId)
+        }
+        
+        AssertAsync {
+            // Assert completion is called
+            Assert.willBeTrue(completionCalled)
+            // Assert message still has `pendingSync` state
+            Assert.willBeEqual(message?.localMessageState, .pendingSync)
+            // Assert message text is updated correctly
+            Assert.willBeEqual(message?.text, updatedText)
+        }
+    }
+    
+    func test_editMessage_updatesMessageLocalStateCorrectly_ifLocalStateIsPendingSend() throws {
```

Comment:
`test_editMessage_updatesMessageLocalStateCorrectly_ifLocalStateIsPendingSend`
`test_editMessage_updatesLocalMessageCorrectly_ifLocalStateIsPendingSync `
`test_editMessage_updatesLocalMessageCorrectly_ifThereIsNoLocalState`

Are almost identical tests, the only thing that changes is the local state condition. Can we make a single test for it and just loop over various local states?